backend/src/api/model/series.rs

```diff
@@ -327,6 +349,193 @@ impl Series {
+        id: Id,
+        title: &str,
+        description: Option<&str>,
+        context: &Context,
+    ) -> ApiResult<Series> {
+        let series = Self::load_for_api(
+            id,
+            context,
+            err::invalid_input!(
+                key = "series.metadata.not-found",
+                "series not found",
+            ),
+            err::not_authorized!(
+                key = "series.metadata.not-allowed",
+                "metadata update not allowed",
+            )
+        ).await?;
+
+        info!(series_id = %id, "Requesting metadata update of series");
+
+        let metadata = serde_json::json!([
+            {
+                "id": "title",
+                "value": title
+            },
+            {
+                "id": "description",
+                "value": description
+            },
+        ]);
+
+        let response = context
+            .oc_client
+            .update_metadata(&series, metadata)
+            .await
+            .map_err(|e| {
+                error!("Failed to send metadata update request: {}", e);
+                err::opencast_unavailable!("Failed to send metadata update request")
+            })?;
+
+        if response.status() == StatusCode::OK {
+            // 200: The series' metadata has been updated.
+            context.db.execute("\
+                update series \
+                set title = $2, description = $3 \
+                where id = $1 \
+            ", &[&series.key, &title, &description]).await?;
+
+            Self::load_by_id(id, context)
+                .await?
+                .ok_or_else(|| err::invalid_input!(
+                    key = "series.metadata.not-found",
+                    "series not found",
+                ))
+        } else {
+            warn!(
+                series_id = %id,
+                "Failed to update series metadata, OC returned status: {}",
+                response.status(),
+            );
+            Err(err::opencast_error!("Opencast API error: {}", response.status()))
+        }
+    }
+
+    pub(crate) async fn delete(id: Id, context: &Context) -> ApiResult<Series> {
+        let series = Self::load_for_api(
+            id,
+            context,
+            err::invalid_input!(
+                key = "series.delete.not-found",
+                "series not found",
+            ),
+            err::not_authorized!(
+                key = "series.delete.not-allowed",
+                "deletion not allowed",
+            )
+        ).await?;
+
+        info!(series_id = %id, "Attempting to send request to delete series in Opencast");
+
+        context.db.execute("\
+            update all_series \
+            set tobira_deletion_timestamp = current_timestamp \
+            where id = $1 \
+        ", &[&series.key]).await?;
+        // Todo: Consider optimistically updating events as well.
+
+
+        let oc_client = context.oc_client.clone();
+        let series_clone = series.clone();
```

Comment:
If this endpoint would return `RemovedSeries` like `Event::delete` used to do, we wouldn't need to clone this, and could remove all the `derive(Clone)`s added in this PR. But ye, performance wise it's pretty whatever, so this is not particularly important.

Reply:
See my [comment](https://github.com/elan-ev/tobira/pull/1313#discussion_r2030095826) on your `RemovedEvent` comment on why this isn't viable right now. (Well depending on who you ask it might be, but not without these relay complaints. There might be some way around those, but I haven't made further investigations.)